compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -358,6 +361,47 @@ struct RewriteExternCallOpToDynamicImportCallOp
   LLVMTypeConverter &typeConverter;
 };
 
+// The 32-bit RISC-V backend is very sensitive to how extended multiplication is
+// lowered. This pattern lowers `arith.mulsi_extended` before going to the LLVM
+// dialect, in a way compatible with that backend, so that we break down any
+// 64-bit constants that would otherwise prevent the code from being vectorized.
+class ExpandMulSIExtended : public OpRewritePattern<arith::MulSIExtendedOp> {
+ public:
+  using OpRewritePattern<arith::MulSIExtendedOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(arith::MulSIExtendedOp op,
+                                PatternRewriter &rewriter) const override {
+    Type resultType = op.getLhs().getType();
+    if (getElementTypeOrSelf(resultType).getIntOrFloatBitWidth() != 32)
+      return failure();
+
+    Location loc = op->getLoc();
```

Comment:
[optional] can we use `op.getLoc()` instead? From my review experience, we use `->` when it is a `Operation *` type or we want to use some methods in `Operation *`.